docs/reference/docs/delete-by-query.asciidoc

```diff
@@ -0,0 +1,308 @@
+[[docs-delete-by-query]]
+== Delete By Query API
+
+experimental[The delete-by-query API is new and should still be considered experimental.  The API may change in ways that are not backwards compatible]
+
+The simplest usage of `_delete_by_query` just performs a deletion on every
```

Comment:
I think maybe you shouldn't be allowed to `_delete_by_query` with specifying a query? `_update_by_query` is less dangerous I think....